diff --git a/libr/core/cmd_print.c b/libr/core/cmd_print.c
index af1d4e3..0c41e5e 100644
--- a/libr/core/cmd_print.c
+++ b/libr/core/cmd_print.c
@@ -1453,7 +1453,10 @@ static int cmd_print(void *data, const char *input) {
 						free (block);
 						pd_result = 0;
 					}
-				} else eprintf ("Cannot find function at 0x%08"PFMT64x"\n", core->offset);
+				} else {
+					eprintf ("Cannot find function at 0x%08"PFMT64x"\n", core->offset);
+					core->num->value = 2;
+				}
 			}
 			break;
 		case 'f': // "pdf"
@@ -1506,6 +1509,7 @@ static int cmd_print(void *data, const char *input) {
 				} else {
 					eprintf ("Cannot find function at 0x%08"PFMT64x"\n", core->offset);
 					processed_cmd = R_TRUE;
+					core->num->value = 2;
 				}
 			}
 			l = 0;
